### PR TITLE
feat: support commands

### LINKED ISSUE #173: Add commands support
Add commands support.

### PR DESCRIPTION
**Pull request by shubham-cmyk**
Add commands support.

Fixes #173

### CODE REVIEW

File: pkg/apis/v1alpha1/command.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package v1alpha1
+
+// Command describes a command to run as a part of a test step or suite.
+type Command struct {
+	// The command and argument to run as a string.
+	Command string `json:"command,omitempty"`
```
Comment (eddycharly) [inline pkg/apis/v1alpha1/command.go:6]: Why `omitempty` ?

File: pkg/apis/v1alpha1/command.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	// If set, the `--namespace` flag will be appended to the command with the namespace to use.
+	Namespaced bool `json:"namespaced,omitempty"`
+	// Ability to run a shell script from TestStep (without a script file)
+	// namespaced and command should not be used with script.  namespaced is ignored and command is an error.
+	// env expansion is depended upon the shell but ENV is passed to the runtime env.
+	Script string `json:"script,omitempty"`
+	// If set, exit failures (`exec.ExitError`) will be ignored. `exec.Error` are NOT ignored.
+	ContinueOnError *bool `json:"continueOnError,omitempty"`
+	// Override the TestSuite timeout for this command (in seconds).
+	Timeout int `json:"timeout,omitempty"`
```
Comment (eddycharly) [inline pkg/apis/v1alpha1/command.go:16]: Should be a pointer and have `// +optional`

File: pkg/apis/v1alpha1/command.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	// The command and argument to run as a string.
+	Command string `json:"command,omitempty"`
+	// If set, the `--namespace` flag will be appended to the command with the namespace to use.
+	Namespaced bool `json:"namespaced,omitempty"`
+	// Ability to run a shell script from TestStep (without a script file)
+	// namespaced and command should not be used with script.  namespaced is ignored and command is an error.
+	// env expansion is depended upon the shell but ENV is passed to the runtime env.
+	Script string `json:"script,omitempty"`
+	// If set, exit failures (`exec.ExitError`) will be ignored. `exec.Error` are NOT ignored.
+	ContinueOnError *bool `json:"continueOnError,omitempty"`
```
Comment (eddycharly) [inline pkg/apis/v1alpha1/command.go:14]: `// +optional`

File: pkg/apis/v1alpha1/test_step_spec.go [hunk 4]
```diff
@@ -33,4 +33,8 @@ type TestStepSpec struct {
 	// This helps in ensuring that the environment is set up correctly before the test step runs.
 	// +optional
 	Delete []Delete `json:"delete,omitempty"`
+
+	// Command provides a list of commands that should be executed as a part of this test step.
+	// +optional
+	Commands []Command `json:"command,omitempty"`
```
Comment (eddycharly) [inline pkg/apis/v1alpha1/test_step_spec.go:39]:
```suggestion
	Command []Command `json:"command,omitempty"`
```

File: pkg/apis/v1alpha1/command.go [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+	// Override the TestSuite timeout for this command (in seconds).
+	Timeout int `json:"timeout,omitempty"`
+	// If set, the output from the command is NOT logged.  Useful for sensitive logs or to reduce noise.
+	SkipLogOutput bool `json:"skipLogOutput,omitempty"`
```
Comment (eddycharly) [inline pkg/apis/v1alpha1/command.go:18]: I think it should be a pointer and also needs `// +optional`

File: pkg/apis/v1alpha1/command.go (hunk 1 above)
Comment (shubham-cmyk) [reply pkg/apis/v1alpha1/command.go:6]: there could be a command or the script 
so I think having a command string is not necessary always

File: pkg/apis/v1alpha1/command.go (hunk 2 above)
Comment (shubham-cmyk) [reply pkg/apis/v1alpha1/command.go:16]: to check if the timeout is empty or not if empty fall back to the default one.

Comment (eddycharly) [reply pkg/apis/v1alpha1/command.go:16]: Should be `Timeout *metav1.Duration`

File: pkg/apis/v1alpha1/command.go (hunk 1 above)
Comment (eddycharly) [reply pkg/apis/v1alpha1/command.go:6]: Hmm, right. How is the script processed ? 🤔

Comment (shubham-cmyk) [reply pkg/apis/v1alpha1/command.go:6]: we just do it like : 
command = exec.CommandContext(ctx, "sh", "-c", cmdStr)

Comment (eddycharly) [reply pkg/apis/v1alpha1/command.go:8]: Ok so it requires a shell...

Comment (eddycharly) [reply pkg/apis/v1alpha1/command.go:8]: We typically build docker images that don't have a shell 🤷

Comment (shubham-cmyk) [reply pkg/apis/v1alpha1/command.go:8]: yes

Comment (shubham-cmyk) [reply pkg/apis/v1alpha1/command.go:8]: > We typically build docker images that don't have a shell 🤷

but how would I exec a shell command now.

File: pkg/apis/v1alpha1/command.go (hunk 5 above)
Comment (shubham-cmyk) [reply pkg/apis/v1alpha1/command.go:24]: why pointer for the `SkipLogOutput` by default we would log and not log that if marked true

Comment (eddycharly) [reply pkg/apis/v1alpha1/command.go:25]: Not necessarily a pointer but must have `// +optional`.

**Comment by shubham-cmyk** [comment]
Ready to merge @eddycharly

**Review by eddycharly — APPROVED** [review]

**Pull request merged**